### PR TITLE
[Main Course] - Ensemble Learning

### PR DESCRIPTION
**Pull request by Tina0330**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Machine_Learning/2022-01-28-Ensemble-Learning.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+###0.936000000000005
+```
+
+### 2.3 Boosting
+
+`Boosting`是指将多个`weak learner`组合在一起的集成方式。与前面几个不同的是，它是按照顺逐个训练分类器，并在每次训练中纠正前一个分类器，最常见的方法就是`Adaptive Boosting(AdaBoost)`和`Gradient Boosting`。
+
+#### 2.3.1 AdaBoost
+
+`AdaBoost`在进化分类器过程中是着重训练欠拟合的训练数据集。如下图所示，在搭建此类分类器时，后续的分类器在训练时会不断地学义以更新前者的权重以提高数据的拟合效果。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Machine_Learning/2022-01-28-Ensemble-Learning.md:119]: 不断地学习

File: docs/Main_Course/Machine_Learning/2022-01-28-Ensemble-Learning.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+)
+
+ada_clf.fit(X_train,y_train)
+
+```
+可通过减少`n_estimator`数量来控制`AdaBoost Ensemble`出现过度拟合的情况。
+
+#### 2.3.2 Gradient Boosting
+
+和`AdaBoosting`一样，梯度提升(Gradient Boosting)也是按照一定序列去学习数据集，不断迭代来生成强壮的集成模型。然而，不同之处在于梯度提升是拟合新数据来减少前者的残差，而非更新前者的权重。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Machine_Learning/2022-01-28-Ensemble-Learning.md:144]: 稳健的集成模型？

**Review by Realvincentyuan — APPROVED** [review]

File: docs/Main_Course/Machine_Learning/2022-01-28-Ensemble-Learning.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+from sklearn.metrics import accuracy_score
+y_pred = bag_clf(X_test)
+accuracy_score(y_test,y_pred)
+
+###0.936000000000005
+```
+
+### 2.3 Boosting
+
+`Boosting`是指将多个`weak learner`组合在一起的集成方式。与前面几个不同的是，它是按照顺逐个训练分类器，并在每次训练中纠正前一个分类器，最常见的方法就是`Adaptive Boosting(AdaBoost)`和`Gradient Boosting`。
```
Comment (jhqian0909) [inline docs/Main_Course/Machine_Learning/2022-01-28-Ensemble-Learning.md:115]: 顺序

File: docs/Main_Course/Machine_Learning/2022-01-28-Ensemble-Learning.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+---
+
+# 集成学习小介
+
+!!! info
+    作者：Tina，发布于2021-06-06，阅读时间：约6分钟，微信公众号文章链接：[:fontawesome-solid-link:]()
+    
+## 1 前言
+
+在机器学习中有一个常见且重要的概念——集成学习（Ensemble Learning），即把多个预测模型的结果集合在一起学习数据集的过程。今天，我们将介绍集成学习的一些常见方法，如`Voting Classifiers`,`Bagging`和`Boosting`。
```
Comment (jhqian0909) [inline docs/Main_Course/Machine_Learning/2022-01-28-Ensemble-Learning.md:12]: 感觉这个定义有一些奇怪

**Review by jhqian0909 — APPROVED** [review]

**Review by EchoTang04 — APPROVED** [review]
LGTM, 感谢Tina分享
